runtime/src/accounts_db.rs

```diff
@@ -3310,15 +3310,18 @@ impl AccountsDb {
         slot: Slot,
         hashes: &[impl Borrow<Hash>],
         mut storage_finder: F,
-        accounts_and_meta_to_store: &[(StoredMeta, &AccountSharedData)],
+        accounts_and_meta_to_store: &[(StoredMeta, Option<&AccountSharedData>)],
     ) -> Vec<AccountInfo> {
         assert_eq!(hashes.len(), accounts_and_meta_to_store.len());
         let mut infos: Vec<AccountInfo> = Vec::with_capacity(accounts_and_meta_to_store.len());
         let mut total_append_accounts_us = 0;
         let mut total_storage_find_us = 0;
         while infos.len() < accounts_and_meta_to_store.len() {
             let mut storage_find = Measure::start("storage_finder");
-            let data_len = accounts_and_meta_to_store[infos.len()].1.data().len();
+            let data_len = accounts_and_meta_to_store[infos.len()]
```

Comment:
if `data_len` is zero, does that mean this account wasn't touched at all? Can we early exit to avoid the storage lookup overhead below?

Reply:
I'm not sure. Looking up the call chain, we decide to store a default AccountSharedData if lamports are 0. This code was pre-existing to my change. I don't know if we allow non-zero lamport, valid accounts with data.len==0. I imagine it is an optimization in the code up the call chain to pass a default account so that we don't store the data[] with a zero-lamport account?